src/lib/ledger/selectTransport.js

```diff
@@ -0,0 +1,15 @@
+module.exports = isBrowser => {
+  if (isBrowser) {
+    const Transport = require('@ledgerhq/hw-transport-webusb').default
+    return Transport
+  } else {
+    if (process.env.NODE_ENV === 'test') {
+      // TODO: replace this with mock hardware interface
+      const Transport = require('@ledgerhq/hw-transport-node-hid').default
+      return Transport
+    } else {
```

Comment:
this doesn't look right to me
mocking should be done from tests code not the application code
by doing it this way, the code in the else branch will never get executed in tests, which means that the tests will not be able to detect bugs in it

Reply:
Yes this is only for testing purposes. The issue in using testnet token address. tokenInfo is not retrieved for testnet token address hence tokenInfo needs to be created and added specifically when running test cases.

The other solution is to use a test specific config file for which has token addresses which will work. For real token addresses tokenInfo will be correct and not null so there will be no need to manually create tokeninfo: https://github.com/DeversiFi/dvf-client-js/blob/9911d0cbe0c844cedd8d657d43ed95d7d41a8978/src/lib/stark/ledger/createSignedOrder.js#L35

Reply:
Having thought about it, I think having a config file for ledger test cases with live token addresses will work well and will not have a need for any test-specific code in the application.